docs/developer-guide/creating-protobuf-schemas.md

```diff
@@ -87,23 +89,56 @@ The Substreams CLI is used to generate the associated Rust code for the protobuf
 {% endhint %}
 
 {% code overflow="wrap" %}
+
 ```bash
 substreams protogen ./substreams.yaml --exclude-paths="sf/ethereum,sf/substreams,google"
 ```
+
 {% endcode %}
 
 The Rust code is generated and saved into [`src/pb/eth.erc721.v1.rs`](https://github.com/streamingfast/substreams-template/blob/develop/src/pb/eth.erc721.v1.rs)``
 
 The [`mod.rs`](https://github.com/streamingfast/substreams-template/blob/develop/src/pb/mod.rs) file located in the `src/pb` directory of the Substreams Template example is responsible for exporting the freshly generated Rust code.
 
 {% code title="src/pb/mod.rs" %}
+
 ```rust
 #[path = "eth.erc721.v1.rs"]
 #[allow(dead_code)]
 pub mod erc721;
 ```
+
 {% endcode %}
 
 View this file in the repo by visiting the following link.
 
 [https://github.com/streamingfast/substreams-template/blob/develop/src/pb/mod.rs](https://github.com/streamingfast/substreams-template/blob/develop/src/pb/mod.rs)
+
+### Protobuf & Rust Optional Fields
+
+Protobufs define fields within them that can be typed as scalar, or simple, values such as a string or integer. Messages can also contain other messages nested within them.
+
+All messages are processed into `Option` enums by `prost`. Any scalar values in a message will return the value assigned, or an empty string, or zero, 0, for integer types. Messages nested within messages will be processed into an `Option`.
+
+The `Option` enum is used to represent the absence of a value in Rust. It allows developers to distinguish between a field that has a value and a field that has not been set. The standard approach to represent nullable data when using Rust is through wrapping optional values in `Option<T>`.
+
+The Rust match keyword is used to compare the value of an `Option` with a `Some` or `None` variant. The Rust `Option` enum is used as the return type for functions, or in this case protobuf-based data fields, that may not always return a value. The following code can be used to check for the existence of a value with match, `Option`, `Some`, and `None`.
+
+```rust
+match person.Location {
+    Some(location) => { // Present, do something },
+    None => { // Not present, do something }
+}
+```
+
+Scenarios that are known to always contain a value can use the `if let` statement and bypass the `None` arm of the `match` code shown above. The following code example demonstrates using `if let` for `Option` fields that are known to always contain a value.
+
+```rust
+if let Some(location) = person.location {
+    // A value is present. The processing logic would be written here.
+}
+```
+
```

Comment:
```suggestion
Scenarios that are known to always contain a value can use the `.unwrap()` call on the `Option` to obtain the wrapped data. This is true if you control the creation of the messages yourself or if the field is documented as always being present

{% hint style=“info” %}
**Note**: Be 100% sure that the field is always present, otherwise your Substreams will panic and will never complete, being stuck on this block forever.
{% endhint %}
```